### DISCUSSION TITLE
Any relevant internal database differences?

### DISCUSSION THREAD

**Discussion by heijligers** [Pythagora-io/gpt-pilot#135] [General]
Hi everybody,
I love the work that has been done on gpt-pilot and think it get's many design choices quite right.

Currently I'm struggling to get gpt-pilot to work in "iterative?" mode. Are there any benefits or differences in using postgress vs sqlite as internal data storage?

Personally I am working on a knowledge integration feature that copies the gpt-pilot internal database into something like memgraph in the hope I can do some optimizing and integration (like exporting and importing code examples and design principles from gpt-pilot to aider and vice versa)

Happy to hear any suggestions or insights
Bjorn

**Comment by lucas-0liveira** [comment]
Hi Bjorn,

It's great to hear that you're enthusiastic about gpt-pilot and the design choices it employs.

Regarding your question about using PostgreSQL versus SQLite as internal data storage, it largely depends on the specific requirements of your project. PostgreSQL is a robust, powerful relational database that supports advanced features and can handle large datasets efficiently. SQLite, on the other hand, is a lightweight, self-contained database that's easy to set up and doesn't require a separate server.

If you're planning to work with large amounts of data or require advanced database features, PostgreSQL might be the better choice. However, if simplicity and ease of deployment are more important to you, SQLite could be a suitable option.

Your idea of integrating gpt-pilot with memgraph for optimizing and integrating knowledge sounds intriguing. It could potentially lead to some innovative solutions.

I would recommend experimenting with both databases and seeing which one better aligns with your specific use case and performance requirements.

Best of luck with your knowledge integration feature, and I hope it proves to be a valuable addition to gpt-pilot!

Cheers,
Lucas

**Comment by YosuCadilla** [comment]
Hi @heijligers 
SQLite, despite the name, is quite capable.
Unless you plan on having a few hundred developers using the same instance of GPT Pilot at the same time, you won't get much benefit from using Postgress.
